### ISSUE TITLE
When executing the same queries, different versions of Neo4j return different results

### ISSUE THREAD

**Issue by abc123000111** [neo4j/neo4j#12852]
I discovered that when I execute the following queries on Neo4j 4.2.15, the result set contains 222 records. But when I execute the same queries on Neo4j 4.4.4, the result set contains 560 records.

```
CREATE (n0) RETURN *;
CREATE (n1 :L1 :L2{k8:''}) RETURN *;
CREATE (n2 :L0) RETURN *;
CREATE (n3 :L0 :L1{k0:-147747132, k10:true}) RETURN *;
CREATE (n4 :L0 :L2) RETURN *;
CREATE (n5 :L3 :L4{k22:false}) RETURN *;
CREATE (n6 :L0{k2:-147747132, k4:'100T'}) RETURN *;
CREATE (n7 :L2{k13:1756231877, k14:true}) RETURN *;
CREATE (n8 :L0{k2:-799239064, k4:'XvE0aZa'}) RETURN *;
CREATE (n9 :L4{k25:true}) RETURN *;
CREATE (n10 :L3 :L4{k20:'7', k21:-943749869}) RETURN *;
CREATE (n11 :L0{k3:true}) RETURN *;
CREATE (n12) RETURN *;
CREATE (n13 :L0 :L2{k13:167164381}) RETURN *;
MATCH (n6 :L0{k2:-147747132, k4:'100T'}), (n2 :L0) MERGE (n6)<-[r0 :T1]-(n2) RETURN *;
MATCH (n0) OPTIONAL MATCH (n0)<--(n1) UNWIND [(n0.k2),0] AS a0 OPTIONAL MATCH (n1),(n2) RETURN 0;
```

**Neo4j Version:** 4.4.4, 4.2.15 Community
**Operating System:** Windows 11
**API:** Cypher / Java API

### Steps to reproduce

1. Start Neo4j 4.2.15 and 4.4.4, respectively.
2. Execute the mentioned queries on each version.
3. Compare the result sets.

### Expected behavior
The result sets are the same.

### Actual behavior
Neo4j 4.2.15: 222 records
Neo4j 4.4.4: 560 records

**Comment by HannesSandberg** [comment]
Thank you! We will take a look!

**Comment by arnefischereit** [comment]
Hi @abc123000111,

This is caused by the same problem as #12818 and a fix will be released with the upcoming 4.4.6 release.
